api/v3/Activity/Getcontactactivities.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/**
+ * Defines the fields for the "getcontactactivities" action api.
+ * The case id, contact id, and return fields are set to be required.
+ *
+ * @param array $params
+ */
+function _civicrm_api3_activity_getcontactactivities_spec(&$params) {
+  _civicrm_api3_activity_get_spec($params);
+
+  $params['case_id']['api.required'] = TRUE;
```

Comment:
It means `case_id` is a required field right? But in the dashboard's activity tab, we show all activities irrespective of the case. Are you sure its working in the dashboard?